zip-0312.rst

```diff
@@ -1,7 +1,556 @@
+      rho_input = rho_input_prefix || G.SerializeScalar(identifier)
+      binding_factor = H1(rho_input)
+      binding_factor_list.append((identifier, binding_factor))
+    return binding_factor_list
+
+
+Round One - Commitment
+''''''''''''''''''''''
+
+Roune One is exactly the same as specified #[FROST]_. But for context, it
+involves these steps:
+
+- Each signer generates nonces and their corresponding public commitments.
+  A nonce is a pair of Scalar values, and a commitment is a pair of Element values.
+- The nonces are stored locally by the signer and kept private for use in the second round.
+- The commitments are sent to the Coordinator.
+
+
+Round Two - Signature Share Generation
+''''''''''''''''''''''''''''''''''''''
+
+In Round Two, the Coordinator generates a random scalar `randomizer` by calling
+`randomizer_generate`. Then it computes `randomizer_commitment = G.ScalarBaseMult(randomizer)`
+and sends it to each signer, over a confidential and authenticated channel,
+along with the message and the set of signing commitments. (Note that this differs
+from regular FROST which just requires an authenticated channel.)
+
+In Zcash, the message that needs to be signed is actually the SIGHASH
+transaction hash, which does not convey enough information for the signers to
+decide if they want to authorize the transaction or not. Therefore, in practice,
+more data is needed to be sent (over the same encrypted, authenticated channel)
+from the Coordinator to the signers, possibly the transaction itself, openings of
+value commitments, decryption of note ciphertexts, etc.; and the signers must check
+that the given SIGHASH matches the data sent from the Coordinator, or compute the
+SIGHASH themselves from that data. However, the specific mechanism for that process
+is outside the scope of this ZIP.
+
+The `sign` function is changed to receive `randomizer_commitment` and incorporate it
+into the computation of the binding factor. It is specified as the following: ::
+
+  Inputs:
+  - identifier, identifier i of the participant, a NonZeroScalar.
+  - sk_i, Signer secret key share, a Scalar.
+  - group_public_key, public key corresponding to the group signing key,
+    an Element.
+  - nonce_i, pair of Scalar values (hiding_nonce, binding_nonce) generated in
+    round one.
+  - msg, the message to be signed, a byte string.
+  - commitment_list =
+      [(j, hiding_nonce_commitment_j, binding_nonce_commitment_j), ...], a
+    list of commitments issued in Round 1 by each participant and sent by the Coordinator.
+    Each element in the list indicates a NonZeroScalar identifier j and two commitment
+    Element values (hiding_nonce_commitment_j, binding_nonce_commitment_j).
+    This list MUST be sorted in ascending order by identifier.
+  - randomizer_commitment, an element in G (sent by the Coordinator).
+
+  Outputs:
+  - sig_share, a signature share, a Scalar.
+
+  def sign(identifier, sk_i, group_public_key, nonce_i, msg, commitment_list):
```

Comment:
(comment to myself to not forget it when I go back to this...)

add `randomizer_commitment`

Reply:
not needed anymore